src/routes/(index)/my-profile/reset-password.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from "react";
+import CanvasCard from "@/components/ui/CanvasCard";
+import { DefaultButton } from "@/components/ui/defaultButton";
+import { createFileRoute } from "@tanstack/react-router";
+import { Input } from "@/components/ui/input";
+import { Typography } from "@/components/ui/typography";
+
+export const Route = createFileRoute("/(index)/my-profile/reset-password")({
+  component: ResetPasswordPage,
+});
+
+function validatePassword(password: string) {
+  // Pelo menos uma letra maiúscula e um número
+  return /[A-Z]/.test(password) && /\d/.test(password);
+}
+
+export default function ResetPasswordPage() {
+  const [password, setPassword] = useState("");
+  const [confirm, setConfirm] = useState("");
+  const [error, setError] = useState("");
+  const [success, setSuccess] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setError("");
+    setSuccess("");
+
+    if (!password || !confirm) {
+      setError("Preencha todos os campos.");
+      return;
+    }
+    if (!validatePassword(password)) {
+      setError("A senha deve conter ao menos uma letra maiúscula e um número.");
+      return;
+    }
+    if (password !== confirm) {
+      setError("As senhas devem ser iguais.");
+      return;
+    }
+
+    setLoading(true);
+    try {
+      // Troque pela rota real de reset
+      const response = await fetch("/api/reset-password", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ password }),
+      });
+      if (response.ok) {
+        setSuccess("Senha redefinida com sucesso!");
+      } else {
+        setError("Não foi possível redefinir a senha. Tente novamente.");
+      }
+    } catch {
+      setError("Erro de conexão. Tente novamente.");
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  return (
+    <div className="flex flex-col bg-background">
+      <div className="flex justify-center pt-16 pb-20">
+        <CanvasCard className="w-full max-w-lg p-8 flex flex-col gap-8 shadow-md">
+          <div className="space-y-4">
+            <Typography className="text-xl font-semibold text-left text-on-banner-text">
+                Redefinir senha
+            </Typography>
+              
+            <div className="h-[1.5px] bg-on-banner-text" />
+          </div>
+          <form onSubmit={handleSubmit} className="space-y-4">
```

Comment:
```suggestion
          <form onSubmit={handleSubmit} className="space-y-4">
```

Atualiza o form aqui com as novas mudanças, pode olhar na branch feature/forgot-password que tem algo parecido